internal/mining/mining.go

```diff
@@ -2177,7 +2177,7 @@ type BgBlkTmplGenerator struct {
 	// notification goroutine so it can in turn asynchronously deliver
 	// notifications to all subscribers.
 	subscriptionMtx   sync.Mutex
-	subscriptions     map[*TemplateSubscription]struct{}
+	subscriptions     map[TemplateSubber]struct{}
```

Comment:
Alright, commenting here because there is the PR where it was changed.  I really don't think this is the right place to do this.  Modifying the mining code does not seem like the right way to go about mocking up these things.

It seems to me that the `TemplateSubber` interface is fine, but it should be defined in `internal/rpcserver/interface.go` alongside `BlockTemplatar` and nothing in `mining` should be modified at all.

Changing out the implementation in the mining code feels wrong to me.

Here is a quick diff I did to show what I mean.  It lacks all the right comments and stuff, but hopefully it illustrates what I'm saying.  Notice that nothing in the mining code is touched, it's very little code, and everything still works for the normal paths.

```
$ git diff
diff --git a/internal/rpcserver/interface.go b/internal/rpcserver/interface.go
index 557faa4c..f7d80d54 100644
--- a/internal/rpcserver/interface.go
+++ b/internal/rpcserver/interface.go
@@ -493,6 +493,11 @@ type CPUMiner interface {
        SetNumWorkers(numWorkers int32)
 }

+type TemplateSubber interface {
+       C() <-chan *mining.TemplateNtfn
+       Stop()
+}
+
 // BlockTemplater represents a source of block templates for use with the
 // RPC server.
 //
@@ -506,7 +511,7 @@ type BlockTemplater interface {
        // template subscription contains functions to retrieve a channel that produces
        // the stream of block templates and to stop the stream when the caller no
        // longer wishes to receive new templates.
-       Subscribe() *mining.TemplateSubscription
+       Subscribe() TemplateSubber

        // CurrentTemplate returns the current template associated with the block
        // templater along with any associated error.
diff --git a/rpcadaptors.go b/rpcadaptors.go
index ad643d7d..7e9c3cca 100644
--- a/rpcadaptors.go
+++ b/rpcadaptors.go
@@ -530,6 +530,10 @@ type rpcBlockTemplater struct {
        *mining.BgBlkTmplGenerator
 }

+func (t *rpcBlockTemplater) Subscribe() rpcserver.TemplateSubber {
+       return t.BgBlkTmplGenerator.Subscribe()
+}
+
 // rpcCPUMiner provides a CPU miner for use with the RPC and implements the
 // rpcserver.CPUMiner interface.
 type rpcCPUMiner struct {
```

With this, inside the `internal/rpcserver/rpcserverhandlers_test.go`file, you can merely make sure your `testBlockTemplater` can merely return a `testTemplateSubber` which implements `rpcserver.TemplaterSubber`.